tests/utils/runner/runner.go

```diff
@@ -0,0 +1,377 @@
+}
+
+// NewNetworkManager constructs a new instance of a network manager
+func NewNetworkManager(config ANRConfig) *NetworkManager {
+	manager := &NetworkManager{
+		ANRConfig: config,
+	}
+
+	logLevel, err := logging.ToLevel(config.LogLevel)
+	if err != nil {
+		panic(fmt.Errorf("invalid ANR log level: %w", err))
+	}
+	manager.logFactory = logging.NewFactory(logging.Config{
+		DisplayLevel: logLevel,
+		LogLevel:     logLevel,
+	})
+
+	return manager
+}
+
+// startServer starts a new ANR server and sets/overwrites the anrServer, done channel, and serverCtxCancel function.
+func (n *NetworkManager) startServer(ctx context.Context) (<-chan struct{}, error) {
+	done := make(chan struct{})
+	zapServerLog, err := n.logFactory.Make("server")
+	if err != nil {
+		return nil, fmt.Errorf("failed to make server log: %w", err)
+	}
+
+	n.anrServer, err = runner_server.New(
+		runner_server.Config{
+			Port:                ":12352",
+			GwPort:              ":12353",
+			GwDisabled:          false,
+			DialTimeout:         10 * time.Second,
+			RedirectNodesOutput: true,
+			SnapshotsDir:        "",
+		},
+		zapServerLog,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to start ANR server: %w", err)
+	}
+	n.done = done
+
+	// Use a separate background context here, since the server should only be canceled by explicit shutdown
+	serverCtx, serverCtxCancel := context.WithCancel(context.Background())
+	n.serverCtxCancel = serverCtxCancel
+	go func() {
+		if err := n.anrServer.Run(serverCtx); err != nil {
+			log.Error("Error shutting down ANR server", "err", err)
+		} else {
+			log.Info("Terminating ANR Server")
+		}
+		close(done)
+	}()
+
+	return done, nil
+}
+
+// startClient starts an ANR Client dialing the ANR server at the expected endpoint.
+// Note: will overwrite client if it already exists.
+func (n *NetworkManager) startClient() error {
+	logLevel, err := logging.ToLevel(n.ANRConfig.LogLevel)
+	if err != nil {
+		return fmt.Errorf("failed to parse ANR log level: %w", err)
+	}
+	logFactory := logging.NewFactory(logging.Config{
+		DisplayLevel: logLevel,
+		LogLevel:     logLevel,
+	})
+	zapLog, err := logFactory.Make("main")
+	if err != nil {
+		return fmt.Errorf("failed to make client log: %w", err)
+	}
+
+	n.anrClient, err = runner_sdk.New(runner_sdk.Config{
+		Endpoint:    "0.0.0.0:12352",
+		DialTimeout: 10 * time.Second,
+	}, zapLog)
+	if err != nil {
+		return fmt.Errorf("failed to start ANR client: %w", err)
+	}
+
+	return nil
+}
+
+// initServer starts the ANR server if it is not populated
+func (n *NetworkManager) initServer() error {
+	if n.anrServer != nil {
+		return nil
+	}
+
+	_, err := n.startServer(context.Background())
+	return err
+}
+
+// initClient starts an ANR client if it not populated
+func (n *NetworkManager) initClient() error {
+	if n.anrClient != nil {
+		return nil
+	}
+
+	return n.startClient()
+}
+
+// init starts the ANR server and client if they are not yet populated
+func (n *NetworkManager) init() error {
```

Comment:
can we rename this to `start()` (or similar) to avoid any possible confusion with built-in `init` function

Reply:
I don't think this should be easily confused with the built-in `init()` function and this follows the convention used in avalanchego to use `init()` to ensure that it is set up correctly before accessing the client/server on the network manager itself.